src/app/frontend/common/components/roleref/component.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, Input} from '@angular/core';
+import {RoleRef} from '@api/backendapi';
+
+@Component({
+  selector: 'kd-role-ref-meta',
+  templateUrl: './template.html',
+})
+export class RoleRefComponent {
+  @Input() initialized = false;
+
+  private roleRef_: RoleRef;
+  get roleRef(): RoleRef {
+    return this.roleRef_;
+  }
+
+  @Input()
+  set roleRef(val: RoleRef) {
+    if (val === undefined) {
+      this.roleRef_ = {} as RoleRef;
+    } else {
+      this.roleRef_ = val;
+    }
+  }
+
+  getClusterRef(name: string): string {
```

Comment:
s/getClusterRef/getClusterRoleHref

Do not hardcode the namespace. It should never be changed this way. The namespace should stay as is when changing views.

Reply:
Yes, I also feel strange beacause when I visit `/clusterrole/xxx?namespace=default`, it's ok but when I visit `/clusterrole/prometheus?namespace=` it will occur some problem, it will show blank, but clusterrole is nothing to do with `namespace`, I wonder if it's caused by where we set the `namespace` to be non empty in someway.

Reply:
Check how other views are doing that.